R/sector_profile_at_product_level.R

```diff
@@ -32,3 +34,54 @@ sp_cols_at_product_level <- function() {
     aka("tsubsector")
   )
 }
+
+if_necessary_fill_grouped_by_with <- function(data, scenarios) {
+  .out <- data |>
+    mutate(
+      all_na = all(is.na(.data$grouped_by)),
+      .by = c(aka("id"), aka("cluster"))
+    )
+
+  no_fill <- filter(.out, all_na)
+  to_fill <- filter(.out, !all_na)
+
+  nothing_to_fill <- identical(nrow(to_fill), 0L)
+  if (nothing_to_fill) {
+    filled <- to_fill
+  } else {
+    filled <- to_fill |> fill_grouped_by_with(scenarios)
+  }
+
+  bind_rows(filled, no_fill) |> select(-all_na)
+}
+
+fill_grouped_by_with <- function(data, scenarios) {
+  levels <- pull(distinct_grouped_by(scenarios))
+  data |>
+    filter(!is.na(.data$grouped_by)) |>
+    group_by(.data[[aka("id")]], .data[[aka("cluster")]]) |>
+    mutate(grouped_by = factor(.data$grouped_by, levels = levels)) |>
+    expand(.data$grouped_by) |>
+    left_join(data, by = c(aka("id"), aka("cluster"), "grouped_by")) |>
```

Comment:
@maurolepore As suggested by you, don't you think you can avoid hard-coding brittle column names like `by = "grouped_by"`?

Reply:
Thanks @kalashsinghal :-)

Yes, I need to cleanup this PR a bit before merging and indeed it would be more maintainable to hide "grouped_by" behind an interface. But in general I'm comfortable using the literal "grouped_by" than "companies_id" or "clustered". 

The reason is that the name "grouped_by" is fully under the control of this package. It's a name in the data structure ("entity") that this package defines to work conveniently with data across all indicators. As such it's way more stable than for example "companies_id" or "clustered" -- which are controlled by external agencies (our managers, the maintainers of tiltIndicatorBefore, etc.).